meeshkan/serve/mock/faker/stateful_faker.py

```diff
@@ -0,0 +1,43 @@
+from http_types import Request
+from openapi_typed_2 import Any, Operation
+
+from meeshkan.serve.mock.faker.stateless_faker import FakerData, StatelessFaker
+from meeshkan.serve.mock.storage.entity import Entity
+from meeshkan.serve.utils.opanapi_utils import get_x
+
+
+class StatefulFaker(StatelessFaker):
+    def fake_ref(self, faker_data: FakerData, schema: Any, depth: int):
+        name = schema["$ref"].split("/")[2]
+        if faker_data.entity is not None and name == faker_data.entity.name:
+            if faker_data.updated_data is not None:
+                return faker_data.updated_data
+            else:
+                return faker_data.entity.query_one(
+                    faker_data.path_item, faker_data.request
+                )
+        else:
+            return self.fake_it(
+                faker_data, faker_data.schema["definitions"][name], depth
+            )
+
+    def fake_ref_array(
+        self, faker_data: FakerData, schema: Any, depth: int, count: int
+    ):
+        name = schema["$ref"].split("/")[2]
+        if faker_data.entity is not None and name == faker_data.entity.name:
```

Comment:
Something like `matches_entity(entity, name)` would maybe make the intention clear here?